src/main/java/org/opensearch/security/state/SecurityClusterManagerListener.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+package org.opensearch.security.state;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.ClusterStateUpdateTask;
+import org.opensearch.cluster.LocalNodeClusterManagerListener;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.Priority;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.configuration.ConfigurationRepository;
+import org.opensearch.threadpool.ThreadPool;
+
+public class SecurityClusterManagerListener implements LocalNodeClusterManagerListener {
+
+    private static final Logger LOGGER = LogManager.getLogger(ConfigurationRepository.class);
+
+    private final ClusterService clusterService;
+
+    private final ConfigurationRepository configurationRepository;
+
+    private final ThreadPool threadPool;
+
+    public SecurityClusterManagerListener(
+        final ClusterService clusterService,
+        final ThreadPool threadPool,
+        final ConfigurationRepository configurationRepository,
+        final Settings settings
+    ) {
+        this.clusterService = clusterService;
+        this.configurationRepository = configurationRepository;
+        this.threadPool = threadPool;
+    }
+
+    @Override
+    public void onClusterManager() {
+        if (!clusterService.state().custom(SecurityMetadata.TYPE, SecurityMetadata.DEFAULT).isSecurityApplied()) {
+            threadPool.executor(ThreadPool.Names.MANAGEMENT).execute(() -> {
+                configurationRepository.initDefaultConfiguration();
```

Comment:
What do you think about adding some way to validate this was completed successful. what about error cases?